src/game_battlealgorithm.cpp

```diff
@@ -518,7 +512,7 @@ bool Game_BattleAlgorithm::Normal::Execute() {
 			hit_chance = weapon.hit;
 			crit_chance = crit_chance += weapon.critical_hit;
 			multiplier = GetAttributeMultiplier(weapon.attribute_set);
-			source->ChangeSp(source->GetSp() - Data::items[weaponID].sp_cost);
+			sp = - Data::items[weaponID].sp_cost;
```

Comment:
This will not work because the "sp = " value is for the SP change of the target.